src/blueprints/tv2/helpers/tv2-show-style-blueprint-configuration-mapper.ts

```diff
@@ -81,14 +87,32 @@ export class Tv2ShowStyleBlueprintConfigurationMapper {
       graphicsSchemas: this.mapGraphicsSchemas(coreConfiguration.GfxSchemaTemplates),
       selectedGraphicsSetup: this.findSelectedGraphicsSetup(coreConfiguration.GfxDefaults, coreConfiguration.GfxSetups),
       splitScreenConfigurations: this.mapSplitScreenConfigurations(coreConfiguration.DVEStyles),
-      breakerTransitionEffectConfigurations: this.mapTransitionEffectConfigurations([
-        ...coreConfiguration.Transitions.map(transition => transition.Transition),
-        coreConfiguration.ShowstyleTransition
-      ]),
+      // SOF-2004 - Need to handle undefined values from Sofie core
+      // breakerTransitionEffectConfigurations: this.mapTransitionEffectConfigurations([
+      //   ...coreConfiguration.Transitions.map(transition => transition.Transition),
+      //   coreConfiguration.ShowstyleTransition
+      // ]),
+      breakerTransitionEffectConfigurations: this.mapBreakerTransitionEffectConfigurations(
+        coreConfiguration.Transitions, 
+        coreConfiguration.ShowstyleTransition),
       breakers: this.mapToBreakers(coreConfiguration.BreakerConfig)
     }
   }
 
+  private mapBreakerTransitionEffectConfigurations(coreTransitions: Transition[], showstyleTransition: string) : BreakerTransitionEffect[] {
+    //SOF-2004 insert guard clause to protect against undefined values
+    if (!coreTransitions) {
+      return []
+    }
+
+    const transitions = coreTransitions.map(transition => transition.Transition).filter(transition => transition !== undefined)
```

Comment:
It's not clear from this line what the type of `transitions` is, so a type declaration should be added.